test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperOutputTest.cs

```diff
@@ -154,7 +115,7 @@ public void GenerateStartTag_ReturnsNothingIfWhitespaceTagName()
             var output = tagHelperOutput.GenerateStartTag();
 
             // Assert
-            Assert.Empty(output);
+            Assert.Null(output);
```

Comment:
@NTaylorMullen the issue we're addressing here isn't _only_ about huge `string`s.  should avoid what unnecessary concatenation or building of new `string`s we can.  e.g. start tags can be composed of loads of individual bits.  in addition, consistency is important.

however, if these methods return `string`s (again, not a good idea), they definitely shouldn't change at all.